packages/enzyme/src/ShallowWrapper.js

```diff
@@ -896,7 +895,8 @@ class ShallowWrapper {
    * @returns {String}
    */
   name() {
-    return this.single('name', displayNameOfNode);
+    const adapter = getAdapter(this[OPTIONS]);
+    return this.single('name', n => adapter.displayNameOfNode(n));
```

Comment:
i think this needs `adapter.displayNameOfNode ? adapter.displayNameOfNode(n) : displayNameOfNode`